service/src/main/java/bio/terra/tanagra/app/AuthInterceptor.java

```diff
@@ -122,10 +124,17 @@ public void postHandle(
       HttpServletRequest request,
       HttpServletResponse response,
       Object handler,
-      ModelAndView modelAndView)
-      throws Exception {
+      ModelAndView modelAndView) {
     // Clear the security context, just to make sure nothing subsequently uses the credentials
     // set up in here.
     SecurityContextHolder.clearContext();
   }
+
+  public static UserId getCurrentUserOrThrow() {
```

Comment:
I am not clear why this method is in the interceptor, although it's not being used by the interceptor. Would it be better to have an AuthUtils or AuthCommon class for these methods instead?

Reply:
I thought it was helpful to have all the auth stuff together, and since this is just one method. But happy to pull this out into a separate class if that's clearer.

Reply:
OK I moved this method to the `UserId` class instead. That class now has 2 static constructors:
- `fromToken` is called from the token utils classes to create the current user object
- `currentUser` is called from the controller classes to get the current user object
PTAL and see if this looks cleaner to you too.

Reply:
thank you @marikomedlock yes this looks much better. I believe that interceptors shouldn't really be referenced by any other service directly, so now things seem clearer.